rasa_core/actions/forms.py

```diff
@@ -89,10 +89,22 @@ def should_request_slot(self, tracker, slot_name):
 
     def get_other_slots(self, tracker):
         requested_slot = tracker.get_slot("requested_slot")
-        slot_events = [f.extract(tracker) for f in self.required_fields()
-                       if isinstance(f, EntityFormField) and
-                       not f.slot_name == requested_slot]
-        return [e for events in slot_events for e in events]
+
+        requested_entity = None
+        for f in self.required_fields():
+            if f.slot_name == requested_slot:
+                requested_entity = getattr(f, 'entity_name', None)
+
+        slot_events = []
+        extracted_entities = {requested_entity}
+
+        for f in self.required_fields():
```

Comment:
this is more of a hotfix to the issue Tom mentioned, right? The first requested field that needs the entity wins, but it might still not be the right one.

Reply:
I think this is fine as a workaround for now but it really doesn't solve the issue.

Reply:
yeah agreed - shall we put it in the inbox to find a better solution for this?

Reply:
already there as entity roles :+1: